### PR TITLE
MDCT-1920: Sort programs on the dashboard by oldest to newest

### PR DESCRIPTION
**Pull request by ntsummers1**
## Description

Fixes [MDCT-1920](https://qmacbis.atlassian.net/browse/MDCT-1920)

I hear the dashboard was ordered from newest to oldest. No longer shall anyone have to suffer at the hands of that sorted depravity. Look upon my code in despair for I have turned the world upside down and now oldest shall be at the top with newest at the bottom.

### How to test
./dev local
Go to the dashboard
Create 2 programs.
The newest program will be at the bottom

...

Want to get crazier? 
Create a 3rd. It'll be at the bottom too.
But wait! `DYNAMO_ENDPOINT=http://localhost:8000 dynamodb-admin` and then go to localhost:8001 or localhost:8000
Click on local-mcpar-reports
Click view on the third report
Change the created at value so that its less then any of the others
Return to the dashboard
See that it is now the oldest and therefor at the top

## Code author checklist
- [x] I have performed a self-review of my code
- [x] I have added [thorough](https://bit.ly/3zPrxuZ) tests
- [x] I have added analytics, if necessary
- [x] I have updated the documentation, if necessary

## Reviewer checklist (two different people)
- [ ] I have done the deep review and verified the items in the above checklist are g2g
- [ ] I have done the lgtm review

### CODE REVIEW

File: services/ui-src/src/utils/reports/reports.test.ts [hunk 1]
```diff
@@ -86,3 +90,39 @@ describe("Test createReportId", () => {
...
+      },
+      {
+        createdAt: 1652568576322,
+        programName: "Newest",
+      },
+    ];
+
+    const sortedReports = [
+      {
+        createdAt: 1652568576322,
```
Comment (ntsummers1) [inline services/ui-src/src/utils/reports/reports.test.ts:113]: Note the difference here is the createdAt Stamp.

File: services/ui-src/src/utils/reports/reports.ts [hunk 2]
```diff
@@ -53,3 +53,14 @@ export const createReportId = (
...
+export const sortReportsOldestToNewest = (
+  reportsArray: { createdAt: number }[]
+) => {
+  return reportsArray.sort(function (
+    stateA: { createdAt: number },
+    stateB: { createdAt: number }
+  ) {
+    return stateA.createdAt - stateB.createdAt;
+  });
+};
```
Comment (braxex) [inline services/ui-src/src/utils/reports/reports.ts:66]: Nit: Could actually get even fancier here by using arrow functions/implicit return and only referencing the createdAt key when needed:

```suggestion
export const sortReportsOldestToNewest = (
  reportsArray: { createdAt: number }[]
) => reportsArray.sort((stateA, stateB) => stateA.createdAt - stateB.createdAt);
```

File: services/ui-src/src/utils/reports/reports.test.ts [hunk 3]
```diff
@@ -86,3 +90,40 @@ describe("Test createReportId", () => {
...
+    const unsortedReports = [
+      {
+        createdAt: 1662568556165,
+        programName: "Second Oldest",
+      },
+      {
+        createdAt: 1662568568589,
+        programName: "Third Oldest",
+      },
+      // Note this createdAt stamp is less than (older) than the prior 2 programs
```
Comment (braxex) [inline services/ui-src/src/utils/reports/reports.test.ts:105]: Nit: Don't necessarily need this comment as the program names should make it obvious. We could also add a readable time to them for future readers' sake, like:

```
{
    createdAt: 1662568556165, // 2022-09-07, 16:35:36 UTC
    programName: "Second Oldest",
},
```

File: services/ui-src/src/components/reports/ReportProvider.tsx [hunk 4]
```diff
@@ -95,7 +96,9 @@ export const ReportProvider = ({ children }: Props) => {
   const fetchReportsByState = async (state: string) => {
     try {
       const result = await getReportsByState(state);
-      setReportsByState(result);
+      // Sort the states returned by the time they were created (Oldest To Newest)
```
Comment (braxex) [inline services/ui-src/src/components/reports/ReportProvider.tsx:99]: Nit: Want to say the reports are sorted, not the states. (Oldest to newest is in the function name so might not need it here).

```suggestion
      // Sort returned reports by time created
```

**Review by braxex — APPROVED** [review]
Looks great! Just a couple comments but nothing blocking.

File: services/ui-src/src/utils/reports/reports.ts (hunk 2 above)
Comment (ntsummers1) [reply services/ui-src/src/utils/reports/reports.ts:66]: condensed!

**Review by gmrabian — APPROVED** [review]
lgtm

**Pull request merged**